src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
```

Comment:
```suggestion
  shape: Shape = 'triangle';

  constructor(public color: Color, public a: number, public b: number, public c: number) {
```
You can add a modifier and typescript will assign values for you
https://www.amitmerchant.com/constructor-assignment-in-typescript/